### ISSUE TITLE
Image size when both MaxWidth and MexHeight are set

### ISSUE THREAD

**Issue by molnarm** [jraps20/Dianoga.ImageMagick#1]
The readme claims that 

> If either Media.Resizing.MaxWidth or Media.Resizing.MaxHeight are set, or if both are set, it will enforce these on the image.

Similarly, in the linked blog post:

> It disallows images to exceed these dimensions, respecting aspect ratio.

However, that doesn't seem to be true, at least not exactly. What I actually see is that if **both** of these restrictions are set, it will shrink the image only until at least **one** of those constraints are met, but the resulting image can still exceed the other.

An example: I have a 3500x2333 image, MaxWidth is set to 1280, MaxHeight is 1024.
Expected output size: 1280x853
Actual output size: 1536x1024

Looking at [the code](https://github.com/jraps20/Dianoga.ImageMagick/blob/master/src/ImageMagickResizer.cs#L27), this seems to be intentional, by using the ^modifiers. It's not really a problem, but it could be mentioned in the documentation.

**Comment by jraps20** [comment]
Thank you for calling out. I do recall adding the ^ modifiers intentionally. I will update all documentation (post and readme) after review.

**Comment by jraps20** [comment]
Commit: #84dc5a39e4d213cf9332ed98c2a2ede5608bb90f

@molnarm after reviewing the code, the `^` was added inadvertently. I removed this modifier from the primary resize logic as you've pointed out.

I verified that it now resizes properly (per the documentation). Whichever dimension is greatest on an image, it will not exceed the specified maximum. I tested in various scenarios:

- Very large landscape images
- Very large portrait images
- Small landscape images
- Small portrait images 

I also tested with various upper bounds: 1920x1080, 1000x1000. In all tests, the image was resized properly. In the scenario where the original image was smaller than the bounds, the image was NOT upscaled, though it's file size was still dramatically reduced.
